app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -127,4 +132,41 @@ def format_sync_error(error)
     result << error.backtrace.join("\n")
     result.mb_chars.limit(ERROR_MAX_SIZE)
   end
+
+  private
+
+  VALID_PLAYBOOK_CHECK = /^\s*?-?\s*?(?:hosts|include|import_playbook):\s*?.*?$/.freeze
+
+  # Confirms two things:
+  #
+  #   - The file extension is a yaml extension
+  #   - The content of the file is "a playbook"
+  #
+  # A file is considered a playbook if it has one line that matches
+  # VALID_PLAYBOOK_CHECK, or it starts with $ANSIBLE_VAULT, which in that case
+  # it is an encrypted file which it isn't possible to decern if it is a
+  # playbook or a different type of yaml file.
+  #
+  def playbook?(filename, worktree)
+    return false unless filename.match?(/\.ya?ml$/)
+
+    content = worktree.read_file(filename)
+    return true if content.start_with?("$ANSIBLE_VAULT")
+
+    content.lines.each do |line|
```

Comment:
Super minor, but it might be better to use `each_line` here vs `lines.each` as the former will not create an intermediate array that is basically a complete copy of the file.

Reply:
Agreed.  👍

Reply:
Fixed.